### ISSUE TITLE
Use `sub-commands` for the command-line / `--help` not listing all sub commands.

### ISSUE THREAD

**Issue by lgautier** [dib-lab/sourmash#102]
The `argparse` module allows the specification of subcommands (https://docs.python.org/dev/library/argparse.html#sub-commands). Using them instead of the current custom solution would:
- facilitate maintenance by relying on the standard library
- make `--help` report all sub-commands (currently not the case - none of the SBT sub-commands is listed).

**Comment by ctb** [comment]
#108 fixes the subcommands that are not listed.

I'm -1 (for now) on actually using argparse subcommands, as that requires that all arguments be available at the time the first parser is built; this would move argument building away from the code that is using the arguments.  There are ways to address this by e.g. making each subcommand its own class, but that seems like overkill.

Nominate to close as #wontfix after #108 is merged.
